src/test/java/teammates/test/pageobjects/InstructorFeedbackResultsPage.java

```diff
@@ -210,20 +210,61 @@ public void clickVisibilityOptionForResponseCommentAndSave(String idString, int
         ThreadHelper.waitFor(1000);
     }
 
-    public boolean verifyAllResultsPanelBodyVisibility(boolean visible) {
-        int numOfQns = browser.driver.findElements(By.cssSelector(".panel-heading+.panel-collapse")).size();
-        assertTrue(numOfQns > 0);
-
-        // Wait for the total duration according to the number of collapse/expand intervals between questions
-        ThreadHelper.waitFor((numOfQns * 50) + 1000);
+    /**
+     * Checks if the body of all the result panels are visible
+     * @return true if all result panel body are visible
+     */
+    public boolean areResultsVisible() {
+        return isAllResultsPanelBodyVisibilityEquals(true);
+    }
 
-        for (WebElement e : browser.driver.findElements(By.cssSelector(".panel-heading+.panel-collapse"))) {
-            if (e.isDisplayed() != visible) {
+    /**
+     * Checks if the body of all the result panels are hidden
+     * @return true if all result panel body are hidden
+     */
+    public boolean areResultsHidden() {
+        return isAllResultsPanelBodyVisibilityEquals(false);
+    }
+    
+    /**
+     * Checks if the body of all the results panels are collapsed or expanded.
+     * @param isVisible true to check for expanded, false to check for collapsed.
+     * @return true if all results panel body are equals to the visibility being checked.
+     */
+    private boolean isAllResultsPanelBodyVisibilityEquals(boolean isVisible) {
+        By panelCollapseSelector = By.cssSelector(".panel-heading+.panel-collapse");
+        List<WebElement> webElements = browser.driver.findElements(panelCollapseSelector);
+        int numOfQns = webElements.size();
+        
+        assertTrue(numOfQns > 0);
+        
+        for (WebElement e : webElements) {
+            if (e.isDisplayed() != isVisible) {
                 return false;
             }
         }
+        
         return true;
     }
+    
+    /**
+     * Method to wait for all the panels to collapse.
+     */    
+    public void waitForPanelsToCollapse() {
+        List<WebElement> panelBodies = browser.driver.findElements(By.cssSelector("div[id^='panelBodyCollapse-']"));
+        waitForElementsVisibility(panelBodies);
+        ThreadHelper.waitFor(1000);
```

Comment:
This wait makes the test slow down quite a lot, and can be improved. Can probably improve this method to check the visibility of the body content of the panel body, or wait for the panel to have the classes "collapse in", or something better. The panelBody becomes visible in the middle of expansion, and isn't necessarily fully expanded yet. The guy who originally implemented this definitely added the waitFor here just so the tests could pass, and meant for this to be fixed later, but then forgot to make an issue for it. :P